plugins/services/src/js/containers/service-detail/ServiceDetail.js

```diff
@@ -182,59 +154,78 @@ class ServiceDetail extends mixin(TabsMixin) {
   getTabs() {
     const {service:{id}} = this.props;
     const routePrefix = `/services/overview/${encodeURIComponent(id)}`;
+    const routeDefinitionPrefix = '/services/overview/:id';
 
     const tabs = [];
-    const activeTab = this.state.currentTab;
+    const {currentTab} = this.state;
 
     tabs.push({
       label: 'Instances',
       callback: () => {
-        this.setState({currentTab: 'tasks'});
+        this.context.router.push(`${routePrefix}/tasks`);
       },
-      isActive: activeTab === 'tasks'
+      isActive: currentTab === `${routeDefinitionPrefix}/tasks`
     });
```

Comment:
Since all of these tabs are routed, you don't need to supply a `callback` method to handle the route changes. Instead you can just pass an array of objects that have the properties `label` and `routePath`. From there, the tabs will use the `Link` component for each tab, which gives us the active indicator for free, so we can remove the `isActive` property and the associated `currentTab` logic.

Check out an example in `NodesTaskDetailPage`: https://github.com/dcos/dcos-ui/blob/master/plugins/nodes/src/js/pages/nodes/NodesTaskDetailPage.js#L36